### PR TITLE
curl: update to 8.8.0

### PR DESCRIPTION
**Pull request by tranzystorekk**
#### Testing the changes
- I tested the changes in this PR: **briefly**

### CODE REVIEW

File: srcpkgs/curl/template [hunk 1]
```diff
@@ -43,6 +43,9 @@ pre_configure() {
 	export CFLAGS="${CFLAGS/-D_FORTIFY_SOURCE=2/}"
 	export CFLAGS="${CFLAGS/-I${XBPS_CROSS_BASE}\/usr\/include/}"
 	export CPPFLAGS="-D_FORTIFY_SOURCE=2"
+
+	# reconfigure for patched changes
+	autoreconf -fi
```
Comment (sgn) [inline srcpkgs/curl/template:48]: Just make man and do install manually!

Comment (Gottox) [reply srcpkgs/curl/template:48]: or we switch to the cmake build?

Comment (tranzystorekk) [reply srcpkgs/curl/template:48]: doing autoreconf felt less invasive to me, as for cmake - I'm not sure what the benefit is when the autotools config is tried and true at this point

Comment (Gottox) [reply srcpkgs/curl/template:48]: I merge as is. The resulting package is the very same.

**Pull request merged**